uavcan/time/62804.Synchronization.1.0.uavcan

```diff
@@ -14,9 +19,9 @@ uint16 MIN_PUBLICATION_PERIOD_ms = 40                   # [millisecond]
 # the switching timeout is 1.5 seconds.
 uint8 PUBLISHER_TIMEOUT_PERIOD_MULTIPLIER = 3
 
-# The time when the PREVIOUS message was transmitted from the current publisher.
+# The time when the PREVIOUS message was transmitted from the current publisher, in microseconds.
 # If this message is published for the first time, this field must be zero.
-Point.1 previous_transmission_timestamp
+truncated uint56 previous_transmission_timestamp_microsecond
```

Comment:
Helps metaprogramming: https://github.com/UAVCAN/dsdl/pull/50#issuecomment-437689685

Reply:
`SynchronizedTimestamp` is also supposed to be used with the diagnostic record message, unless we don't want any timestamping there (questionable). So it's not only about SI. I prefer to keep everything time-related that is sufficiently generic under this namespace to simplify reuse in vendor-specific type definitions.

If we chose to use overflowing timestamps, then 32-bit is our only option because 24-bit overflows too frequently, and 26-bit is not byte-aligned (and it overflows too frequently, too). If we ended up wanting overflowing timestamps, we shall add another type here, `SynchronizedOverflowingTimestamp` or something similar; we must keep non-overflowing timestamp as well for completeness and for encouraging safer designs ("safer" here means designs that don't trade-off bandwidth for time uncertainty; in some applications this can be preferred).

Reply:
Also, to answer your first question in this thread: the main reason I don't use SynchronizedTimestamp in this message is that this field is not actually a synchronized timestamp in its usual sense. It is the timestamp of the moment when the previous message was transmitted, so I used a different type to make this more explicit.